api/src/main/java/org/geysermc/geyser/api/item/custom/NonVanillaCustomItemData.java

```diff
@@ -162,11 +162,11 @@ default boolean isTool() {
     }
 
     /**
-     * Gets the block the item places.
+     * Gets the block placer settings, if it's null then the component won't be added.
      *
-     * @return the block the item places
+     * @return the block placer settings
      */
-    String block();
```

Comment:
Hi, thanks for the PR!

Because this is API, we can't delete old methods without older extensions possibly breaking expecting the method. Let's deprecate this (and the builder method) and redirect it to create a CustomBlockPlacer with replaceBlockItem enabled/disabled.

Also, if you're able to revert the formatting changes in CustomItemRegistryPopulator, that would be nice just for diff viewing.